pom.xml

```diff
@@ -29,6 +29,10 @@
     <hamcrest.version>1.3</hamcrest.version>
     <maven-shade-plugin.version>3.0.0</maven-shade-plugin.version>
     <relocation-prefix>com.hotels.hcommon.ssh</relocation-prefix>
+    <guava.version>20.0</guava.version>
```

Comment:
Why this version of Guava? Also, since we're only using the Joiner I'd argue we don't add this dependency and just use standard JDK to do the string joining. It will make using this downstream in other projects less likely to have classpath clashes (since this is a library class it's something we need to keep in mind and try keep the dependencies to a minimum). Other option would be to shade and relocate it in.

Reply:
`String.join()` is available only for Java 8+, right? This is using Java 7. Would it be better to use StringUtils from Apache instead?

Reply:
And also, any Guava that is 21+ gives me this kind of error:
`java.lang.UnsupportedClassVersionError: com/google/common/base/Joiner : Unsupported major.minor version 52.0`

Reply:
So Guava 21 then must require Java 8. As for Apache StringUtils - it just moves the problem to a different library, the only slight benefit with it is that the versions of that change far less frequently so its less likely to cause problems downstream. One option would be to look at the Guava and StringUtils code and if it's like one or two methods we could just pull them in here (with attrribution of course) and then put a TODO to replace it all with String.join when we move this project to Java 8. Normally I'd prefer doing what you've done (i.e. use someone else's library) but we've been bitten quite a bit in the projects downstream of this with Guava version clashes.